### PR TITLE
samurai : new package

### PR DESCRIPTION
**Pull request by gouarin**
Specify library name and version:  **samurai/0.2.0**

I'm the author of this library and I would like to see it in conan center.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by RubenRBS — COMMENTED** [review]
Hi @gouarin thanks a lot for your time, we really appreciate it. I'm reviewing this now, sorry about getting so processing our way thru the PR backlog :)
Let me know if you need help getting thus thru the finish line

File: recipes/samurai/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    no_copy_source = True
+
+    @property
+    def _min_cppstd(self):
+        return "17"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16.9",
```
Comment (RubenRBS) [inline recipes/samurai/all/conanfile.py:33]: Where does this number come from? Afaik c++std 17 was not supported in visual studio until 19.0

File: recipes/samurai/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            "msvc": "192",
+            "gcc": "12",
+            "clang": "13",
+            "apple-clang": "14",
```
Comment (RubenRBS) [inline recipes/samurai/all/conanfile.py:37]: Does it need such recent versions?

File: recipes/samurai/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+        # In case this library does not work in some another configuration, it should be validated here too
+        # if self.settings.os == "Windows":
+        #     raise ConanInvalidConfiguration(f"{self.ref} can not be used on Windows.")
```
Comment (RubenRBS) [inline recipes/samurai/all/conanfile.py:65]: Feel free to remove unused code from the template that just adds noise :)

File: recipes/samurai/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    def package_info(self):
+        # Folders not used for header-only
+        self.cpp_info.bindirs = []
+        self.cpp_info.libdirs = []
+
+        # Set these to the appropriate values if the package has an official FindPACKAGE.cmake
+        # listed in https://cmake.org/cmake/help/latest/manual/cmake-modules.7.html#find-modules
+        # examples: bzip2, freetype, gdal, icu, libcurl, libjpeg, libpng, libtiff, openssl, sqlite3, zlib...
+        self.cpp_info.set_property("cmake_module_file_name", "samurai")
+        self.cpp_info.set_property("cmake_module_target_name", "samurai::samurai")
```
Comment (RubenRBS) [inline recipes/samurai/all/conanfile.py:89]:
```suggestion
        # Set these to the appropriate values if the package has an official FindPACKAGE.cmake
        # listed in https://cmake.org/cmake/help/latest/manual/cmake-modules.7.html#find-modules
        # examples: bzip2, freetype, gdal, icu, libcurl, libjpeg, libpng, libtiff, openssl, sqlite3, zlib...
        self.cpp_info.set_property("cmake_module_file_name", "samurai")
        self.cpp_info.set_property("cmake_module_target_name", "samurai::samurai")
```

Does not seem to be necessary, right? As the comment above states, this is only needed if an official FindPackage exists

File: recipes/samurai/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy
```
Comment (RubenRBS) [inline recipes/samurai/all/conanfile.py:4]:
```suggestion
from conan.tools.files import get, copy
```

**Review by RubenRBS — APPROVED** [review]

**Review by RubenRBS — CHANGES_REQUESTED** [review]
Didn't mean to approve it just yet :)

File: recipes/samurai/all/test_v1_package/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 3.15)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
```
Comment (memsharded) [inline recipes/samurai/all/test_v1_package/CMakeLists.txt]: The test_v1_package is no longer necessary

File: recipes/samurai/all/conanfile.py (hunk 4 above)
Comment (memsharded) [reply recipes/samurai/all/conanfile.py]: I am cleaning it in my commit

File: recipes/samurai/all/test_v1_package/CMakeLists.txt (hunk 6 above)
Comment (memsharded) [reply recipes/samurai/all/test_v1_package/CMakeLists.txt]: Removing it in my commit

File: recipes/samurai/all/conanfile.py (hunk 3 above)
Comment (memsharded) [reply recipes/samurai/all/conanfile.py]: removed

File: recipes/samurai/all/conanfile.py (hunk 2 above)
Comment (memsharded) [reply recipes/samurai/all/conanfile.py:35]: Apparently yes, approving to be merged.

File: recipes/samurai/all/conanfile.py (hunk 1 above)
Comment (memsharded) [reply recipes/samurai/all/conanfile.py:31]: It is ok, /c++17 exists from earlier

**Review by memsharded — APPROVED** [review]

**Comment by gouarin** [comment]
Thanks @memsharded!

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**